packages/enzyme-adapter-utils/src/Utils.js

```diff
@@ -153,6 +153,10 @@ export function flatten(arrs) {
   return flatArrs;
 }
 
+export function ensureKeyOrUndefined(key) {
+  return key || (key === '' ? '' : undefined);
```

Comment:
This check could be more restrictive.
```js
return typeof key === 'string' ? key : undefined;
```
Are there situations where a key can be anything other than a string or a falsey value?

Reply:
react stringifies all valid keys before storing them, and a key is valid if it is not undefined. So i think this workes either way. I think it's probably best to leave it as you have now

Reply:
maybe `typeof key === 'string' ? key : undefined`?